src/avfoundation.cs

```diff
@@ -10868,4 +10868,131 @@ interface AVContentProposal : NSCopying
 		IntPtr Constructor (CMTime contentTimeForTransition, string title, [NullAllowed] UIImage previewImage);
 	}
 #endif
+
+	[TV (10,2), Mac (10,12,4), iOS (10,3)]
+	[Protocol, Model]
+	[BaseType (typeof (NSObject))]
+	interface AVContentKeySessionDelegate
+	{
+		[Abstract]
+		[Export ("contentKeySession:didProvideContentKeyRequest:")]
+		void DidProvideContentKeyRequest (AVContentKeySession session, AVContentKeyRequest keyRequest);
+
+		[Export ("contentKeySession:didProvideRenewingContentKeyRequest:")]
+		void DidProvideRenewingContentKeyRequest (AVContentKeySession session, AVContentKeyRequest keyRequest);
+
+		[Export ("contentKeySession:didProvidePersistableContentKeyRequest:")]
+		void DidProvidePersistableContentKeyRequest (AVContentKeySession session, AVPersistableContentKeyRequest keyRequest);
+
+		[Export ("contentKeySession:contentKeyRequest:didFailWithError:")]
+		void DidFail (AVContentKeySession session, AVContentKeyRequest keyRequest, NSError err);
+
+		[Export ("contentKeySession:shouldRetryContentKeyRequest:reason:")]
+		bool ShouldRetryContentKeyRequest (AVContentKeySession session, AVContentKeyRequest keyRequest, string retryReason);
+
+		[Export ("contentKeySessionContentProtectionSessionIdentifierDidChange:")]
+		void DidChange (AVContentKeySession session);
+	}
+
+	partial interface IAVContentKeySessionDelegate {}
+
+	[TV (10,2), Mac (10,12,4), iOS (10,3)]
+	[BaseType (typeof (NSObject))]
+	interface AVContentKeySession {
+		[Static]
+		[Export ("contentKeySessionWithKeySystem:storageDirectoryAtURL:")]
+		AVContentKeySession Create (string keySystem, [NullAllowed] NSUrl storageURL);
+
+		[Export ("setDelegate:queue:")]
+		void SetDelegate ([NullAllowed] IAVContentKeySessionDelegate newDelegate, [NullAllowed] DispatchQueue delegateQueue);
+
+		[Wrap ("WeakDelegate")]
+		[NullAllowed]
+		IAVContentKeySessionDelegate Delegate { get; }
+
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Weak)]
+		NSObject WeakDelegate { get; }
```

Comment:
Since the `Delegate` property is defined using an interface, you can completely remove the `WeakDelegate` (and put the `Export` attribute on the `Delegate` property instead).